controllers/capability/reconciler.go

```diff
@@ -174,15 +170,17 @@ func (r *Reconciler) deleteStatefulSetIfOldLabelsAreUsed(desiredSts *appsv1.Stat
 		return false, err
 	}
 
-	if _, ok := currentSts.Labels[OldKeyActiveGate]; !ok {
-		return false, nil
+	for k := range desiredSts.Labels {
```

Comment:
Alternatively, you can go with `reflect.DeepEqual(currentSts.Labels, desiredSts.Labels)`

Reply:
Ok, thx

Did not know that